### ISSUE TITLE
Shatterline (2087030)

### ISSUE THREAD

**Issue by xpander69** [ValveSoftware/Proton#6165]
# Compatibility Report
- Name of the game with compatibility issues: Shatterline
- Steam AppID of the game: 2087030

## System Information
- GPU: RTX 3080
- Driver/LLVM version:515.49.14
- Kernel version: 5.19-xanmod
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/xpander69/ac99bf12ebfb90801534c30be40a13c5
- Proton version: proton-experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-2087030.log](https://github.com/ValveSoftware/Proton/files/9586800/steam-2087030.log)

## Symptoms 
Textures doesn't seem to load. Everything is rendered dark. Low and Highest settings.
https://i.imgur.com/XYa3J32.png

## Reproduction
1. Launch the game. Throws you into tutorial map first, where the issue can already be seen.

**Comment by xpander69** [comment]
_installing d3dcompiler_47 fixes the dark texture issue_ Not needed since R66 game update
setting prefix to win 7 fixes the connection error

those are the workarounds to make the game playable right now

`protontricks 2087030 win7`

updated.

**Comment by Kreyren** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Shatterline
- Steam AppID of the game: 2087030

## System Information
- OS: GNU Guix GNU/Linux
```console
$ guix describe
Generation 4	Sep 22 2022 07:58:11	(current)
  nonguix 415aeda
    repository URL: https://gitlab.com/nonguix/nonguix
    branch: master
    commit: 415aeda016ca174764a700d8493cc93285ebe3dd
  guix 04cad8e
    repository URL: https://git.savannah.gnu.org/git/guix.git
    branch: master
    commit: 04cad8e29ed85be838921c7fa05f7c5bce94b6d1
```
- Arch: x86_64
- Model: Z68A-D3-B3
- CPU: i7-2680K
- GPU: AMD RX 570
- Driver/LLVM version: AMDGPU (Linux 5.19.8)
	- LLVM: 11.0.0
	- DRM: 3.47.0
	- VBIOS version 113V34122-F3
	- Arch: POLARIS10
	- Mesa: 21.3.8
-  Drivers in flatpak: Driver Version:  4.6 (Compatibility Profile) Mesa 22.1.7 (git-7a3414c4bd)
- Kernel version: GNU/Linux 5.19.8
- Window Manager: GNOME Shell
- X Server Release: 12101004
- 
- Link to full system information report as [Gist](https://gist.github.com/): No
- Proton version: Experimental and 7.0-4 tried

## I confirm:
- [NO] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.



After verifying game files this is all it gives me:

```console
$ cat ~/.var/app/com.valvesoftware.Steam/steam-2087030.log 
======================
Proton: 1660576899 proton-7.0-4
SteamGameId: 2087030
Command: ['/home/kreyren/.var/app/com.valvesoftware.Steam/.local/share/Steam/steamapps/common/Shatterline/Launch_Game.exe', '-platform_type', 'Steam', '-bugsplat_enabled']
Options: {'forcelgadd'}
depot: 0.20220919.70
pressure-vessel: 0.20220919.0
scripts: v0.20220823.0-0-gcc4e44f
soldier: 0.20220919.0 soldier 0.20220919.0
Kernel: Linux 5.19.8 #1 SMP PREEMPT_DYNAMIC 1 x86_64
======================
```

## Symptoms 
The game install**ed** fine and ran great (platinum) with d3dcompiler_47 and win7 until 22.09.22 when it got updated and now i can't launch it after checking file integrity and before i couldn't join any match (likely anti-cheat)

1. Failure to load textures resulting in texture-less environment:

Fixed by adding `d3dcompiler_47` though winetricks

2. Unable to join any game:

Originally fixed by setting the protonprefix on Windows 7 until 22.09.22 where game got updated **and it's broken again**

3. On bordless fullscreen window sometimes after the game loads into the match it causes me to be unable to move excluding looking around with mouse (unable to shoot) and unable to change window so i have to always change in TTY and kill it in `htop`, hotfix to this was to play the game in windowed.

## Reproduction

1. Get Steam from Flatpak on any GNU distro with Linux kernel
2. Install Shatterline on steam
3. Try to play the game

## Additional notes
* Some people on protondb are complaining about shutter during gameplay which is not present on my setup

**Comment by Kreyren** [comment]
# Compatibility Report
- Name of the game with compatibility issues: Shatterline
- Steam AppID of the game: 2087030

## System Information
- OS: GNU Guix GNU/Linux
```console
$ guix describe
Generation 5	Sep 24 2022 07:56:35	(current)
  nonguix e49878a
    repository URL: https://gitlab.com/nonguix/nonguix
    branch: master
    commit: e49878a50241ded03b9b31d6d6995476577ea1ca
  guix a5c876c
    repository URL: https://git.savannah.gnu.org/git/guix.git
    branch: master
    commit: a5c876cdb2d9ab79ba5027ffad7893dafbece50b
```
- Arch: x86_64
- Model: Z68A-D3-B3
- CPU: i7-2680K
- GPU: AMD RX 570
- Driver/LLVM version: AMDGPU (Linux 5.19.8)
	- LLVM: 11.0.0
	- DRM: 3.47.0
	- VBIOS version 113V34122-F3
	- Arch: POLARIS10
	- Mesa: 21.3.8
- Kernel version: GNU/Linux 5.19.8
- Window Manager: GNOME Shell
- X Server Release: 12101004
- Link to full system information report as [Gist](https://gist.github.com/): No
- Proton version: Experimental and 7.0-4 tried

## I confirm:
- [NO] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.



Proton7.0-4 (without win7 and d3dcompiler): https://gist.github.com/Kreyren/0d1eacd4fe199b11c468a9cd7a53636e

<details><summary>Click to expand</summary><p>

```
$ WINEPREFIX=~/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx winetricks d3dcompiler_47 win7
Executing mkdir -p /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
------------------------------------------------------
WINEPREFIX INFO:
Drive C: total 4
drwxr-xr-x  9 kreyren users  131 24. zář 12.44 .
drwxr-xr-x  4 kreyren users  117 24. zář 12.46 ..
drwxr-xr-x  2 kreyren users   31 24. zář 12.44 openxr
drwxr-xr-x  4 kreyren users   44 24. zář 12.44 ProgramData
drwxr-xr-x  6 kreyren users   97 24. zář 12.44 Program Files
drwxr-xr-x  8 kreyren users  135 24. zář 12.44 Program Files (x86)
drwxr-xr-x  4 kreyren users   37 24. zář 12.44 users
drwxr-xr-x  3 kreyren users   17 24. zář 12.44 vrclient
drwxr-xr-x 21 kreyren users 4096 24. zář 12.44 windows

Registry info:
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/system.reg:#arch=win64
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/userdef.reg:#arch=win64
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/user.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
warning: wine cmd.exe /c echo '%AppData%' returned empty string, error message "wine: could not load kernel32.dll, status c000000f
wine: could not load kernel32.dll, status c000000f" 
------------------------------------------------------
kreyren@dreamon /home/kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata$ WINEPREFIX=~/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx winetricks d3dcompiler_47 win7
Executing mkdir -p /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
------------------------------------------------------
WINEPREFIX INFO:
Drive C: total 4
drwxr-xr-x  9 kreyren users  131 24. zář 12.44 .
drwxr-xr-x  4 kreyren users  117 24. zář 12.46 ..
drwxr-xr-x  2 kreyren users   31 24. zář 12.44 openxr
drwxr-xr-x  4 kreyren users   44 24. zář 12.44 ProgramData
drwxr-xr-x  6 kreyren users   97 24. zář 12.44 Program Files
drwxr-xr-x  8 kreyren users  135 24. zář 12.44 Program Files (x86)
drwxr-xr-x  4 kreyren users   37 24. zář 12.44 users
drwxr-xr-x  3 kreyren users   17 24. zář 12.44 vrclient
drwxr-xr-x 21 kreyren users 4096 24. zář 12.44 windows

Registry info:
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/system.reg:#arch=win64
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/userdef.reg:#arch=win64
/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/user.reg:#arch=win64
------------------------------------------------------
------------------------------------------------------
warning: wine cmd.exe /c echo '%AppData%' returned empty string, error message "wine: could not load kernel32.dll, status c000000f
wine: could not load kernel32.dll, status c000000f" 
------------------------------------------------------
$ rm -r /home/kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx
$ WINEPREFIX=~/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx winetricks d3dcompiler_47 win7
Executing mkdir -p /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Using winetricks 20220411 - sha256sum: fa32f0975aa4733e5e9ac8b12b23e2c70fcc1261f9c865b79704ae3cc27541ca with wine-7.8 (Staging) and WINEARCH=win64
Executing w_do_call d3dcompiler_47
Executing mkdir -p /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing load_d3dcompiler_47 
Executing 7z x /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win32.exe -o/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/temp/win32 core/d3dcompiler_47.dll

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.utf8,Utf16=on,HugeFiles=on,64 bits,8 CPUs Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz (206A7),ASM,AES-NI)

Scanning the drive for archives:
1 file, 37111376 bytes (36 MiB)                           

Extracting archive: /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win32.exe
--       
Path = /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win32.exe
Type = PE
Physical Size = 37111376
CPU = x86
Characteristics = Executable 32-bit NoRelocs NoLineNums NoLocalSyms
Created = 2018-04-26 17:54:09
Headers Size = 4096
Checksum = 37158920
Name = 7zS.sfx.exe
Image Size = 274432
Section Alignment = 4096
File Alignment = 512
Code Size = 57344
Initialized Data Size = 65536
Uninitialized Data Size = 147456
Linker Version = 6.0
OS Version = 4.0
Image Version = 0.0
Subsystem Version = 4.0
Subsystem = Windows GUI
Stack Reserve = 1048576
Stack Commit = 4096
Heap Reserve = 1048576
Heap Commit = 4096
Image Base = 4194304
Comment = FileVersion: 18.1.0.0
FileVersion: 18.01
ProductVersion: 18.1.0.0
ProductVersion: 18.01
CompanyName: Igor Pavlov
FileDescription: 7z Setup SFX
InternalName: 7zS.sfx
LegalCopyright: Copyright (c) 1999-2018 Igor Pavlov
OriginalFilename: 7zS.sfx.exe
ProductName: 7-Zip
----
Path = [0]
Size = 36983424
Packed Size = 36983424
Virtual Size = 36983424
Offset = 120320
--
Path = [0]
Type = 7z
Offset = 81
Physical Size = 36983337
Tail Size = 6
Headers Size = 2137
Method = LZMA:25 BCJ2
Solid = +
Blocks = 1

Everything is Ok              

Files: 61
Size:       46133485
Compressed: 37111376
Executing cp -f /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/temp/win32/core/d3dcompiler_47.dll /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/syswow64/d3dcompiler_47.dll
Executing 7z x /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win64.exe -o/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/temp/win64 core/d3dcompiler_47.dll

7-Zip [64] 16.02 : Copyright (c) 1999-2016 Igor Pavlov : 2016-05-21
p7zip Version 16.02 (locale=en_US.utf8,Utf16=on,HugeFiles=on,64 bits,8 CPUs Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz (206A7),ASM,AES-NI)

Scanning the drive for archives:
1 file, 40176024 bytes (39 MiB)                           

Extracting archive: /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win64.exe
--       
Path = /home//kreyren/.cache/winetricks/d3dcompiler_47/FirefoxSetup62.0.3-win64.exe
Type = PE
Physical Size = 40176024
CPU = x86
Characteristics = Executable 32-bit NoRelocs NoLineNums NoLocalSyms
Created = 2018-04-26 17:54:09
Headers Size = 4096
Checksum = 40199636
Name = 7zS.sfx.exe
Image Size = 274432
Section Alignment = 4096
File Alignment = 512
Code Size = 57344
Initialized Data Size = 65536
Uninitialized Data Size = 147456
Linker Version = 6.0
OS Version = 4.0
Image Version = 0.0
Subsystem Version = 4.0
Subsystem = Windows GUI
Stack Reserve = 1048576
Stack Commit = 4096
Heap Reserve = 1048576
Heap Commit = 4096
Image Base = 4194304
Comment = FileVersion: 18.1.0.0
FileVersion: 18.01
ProductVersion: 18.1.0.0
ProductVersion: 18.01
CompanyName: Igor Pavlov
FileDescription: 7z Setup SFX
InternalName: 7zS.sfx
LegalCopyright: Copyright (c) 1999-2018 Igor Pavlov
OriginalFilename: 7zS.sfx.exe
ProductName: 7-Zip
----
Path = [0]
Size = 40048072
Packed Size = 40048072
Virtual Size = 40048072
Offset = 120320
--
Path = [0]
Type = 7z
Offset = 81
Physical Size = 40047984
Tail Size = 7
Headers Size = 2125
Method = LZMA:25 BCJ2
Solid = +
Blocks = 1

Everything is Ok                 

Files: 60
Size:       46848324
Compressed: 40176024
Executing cp -f /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/temp/win64/core/d3dcompiler_47.dll /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx/dosdevices/c:/windows/system32/d3dcompiler_47.dll
Using native override for following DLLs: d3dcompiler_47
Executing wine regedit C:\windows\Temp\override-dll.reg
Executing wine64 regedit C:\windows\Temp\override-dll.reg
Executing w_do_call win7
Executing mkdir -p /home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030
------------------------------------------------------
warning: You are using a 64-bit WINEPREFIX. Note that many verbs only install 32-bit versions of packages. If you encounter problems, please retest in a clean 32-bit WINEPREFIX before reporting a bug.
------------------------------------------------------
Executing load_win7 
Executing wine winecfg -v win7
------------------------------------------------------
warning: Running /home//kreyren/.guix-profile/bin/wineserver -w. This will hang until all wine processes in prefix=/home//kreyren/.local/share/guix-sandbox-home/.local/share/Steam/steamapps/compatdata/2087030/pfx terminate
------------------------------------------------------
```

</p></details>

Proton-7.0-4 with changes above: https://gist.github.com/Kreyren/cdabedf77bfb17f2b38acf8e12256c71

Proton-experimental: https://gist.github.com/Kreyren/3233aa98d88057b8fb08b6985e0ab25b

## Symptoms 
The game install**ed** fine and ran great (platinum) with d3dcompiler_47 and win7 until 22.09.22 where i can no longer join any public match

1. Failure to load textures resulting in texture-less environment:

Fixed by adding `d3dcompiler_47` though winetricks

2. Unable to join any game:

Originally fixed by setting the protonprefix on Windows 7 until 22.09.22 where game got updated **and it's broken again**

3. On borderless fullscreen window sometimes after the game loads into the match it causes me to be unable to move excluding looking around with mouse (unable to shoot) and unable to change window so i have to always change in TTY and kill it in `htop`, hotfix to this was to play the game in windowed.

## Reproduction

0. Get any GNU distro with Linux kernel that has GNU Guix packaged
1. Install GNU Guix e.g. on apt-based distributions `# apt-get install guix`
2. Get the following channels:
```
$ cat ~/.config/guix/channels.scm
(list
 (channel
  (name 'guix)
  (url "https://git.savannah.gnu.org/git/guix.git")
  (branch "master")
  (introduction
   (make-channel-introduction
    "9edb3f66fd807b096b48283debdcddccfea34bad"     ;2020-05-26
    (openpgp-fingerprint                           ;mbakke
     "BBB0 2DDF 2CEA F6A8 0D1D  E643 A2A0 6DF2 A33A 54FA"))))
 (channel
  (name 'nonguix)
  (url "https://gitlab.com/nonguix/nonguix")
  (introduction
   (make-channel-introduction
    "897c1a470da759236cc11798f4e0a5f7d4d59fbc"
    (openpgp-fingerprint
     "2A39 3FFF 68F4 EF7A 3D29  12AF 6F51 20A0 22FB B2D5")))))
```
3. Update repositories -- `$ guix pull`
4. Open steam from guix -- `$ guix shell steam -- steam`
5. Install Shatterline and try to play the game

## Additional notes
* Some people on protondb are complaining about shutter during gameplay which is not present on my setup

**Comment by DarkestByte** [comment]
> The game installed fine and ran great (platinum) with d3dcompiler_47 and win7 until 22.09.22 where i can no longer join any public match

this is the update that broke it: https://store.steampowered.com/news/app/2087030/view/3395173894699741120
after that a friend and I could no longer join matches.

**Comment by Kreyren** [comment]
@kisak-valve I talked with the developer and they seems to be willing to enable linux support, but don't understand what proton is, how it works and how it would affect the development (they are startup-ish so their development resources are limited etc..) and how it would make the game cheatable bcs they currently have issues with cheating in the game.

Could you get someone from valve to e.g. write some short and sensible thing here and how to configure EAC so that the game works again?

EDIT: Was given https://partner.steamgames.com/doc/steamdeck/proton#1 which was relayed to the developer

**Comment by Kreyren** [comment]
I helped the developer to fix the game and new update is scheduled to land within 24h hours to make the game work for all linux users. ^-^

The linux module for EOS is enabled since 23rd Aug 2022:
![shatterline-linux-eac-module](https://user-images.githubusercontent.com/11302521/192392346-6110b702-3272-40b7-a768-5bee3e0c1943.png)

The developer did setup according to the provided documentation in https://partner.steamgames.com/doc/steamdeck/proton#1 and noted the requirement to update the linux module.

There was a bug in the pipeline which caused the public release of the game to miss `Bin/CrashRpt1403.dll` which was triggering the EOS resulting in linux players being unable to join any public game lobby the solution is to provide the file. Discovered through searching in `Steam/steamapps/compatdata/2087030/pfx/drive_c/users/steamuser/Saved Games/Shatterline/somerandomchars/user/log/Game.log` which included:

> <2022-09-26T22:09:54:789+02> [Warning] (Game) - [EOSCore_Windows::EOSSDKLoggingCallback] LogEOSAntiCheat: [AntiCheatClient] [PollStatusInternal] Client Violation with Type: 5 Message: Unknown file version (Bin/CrashRpt1403.dll), len: 44

The `d3dcompiler_47.dll` **was required** as they broke shaders for proton which will be fixed after the game realease and that currently results in a black textures when you try to join the game, with new update that dependency is provided by upstream so **it's no longer required to workaround it** and workarounds will trigger EOS.

The `win7` workaround is still needed as the game is using **Winsock Registered Input/Output (RIO) API** that is not implemented in wine/proton if the windows release is set on 8 and higher, developer plans to address it before the game gets released or shortly after by replacing it with proton-friendly alternative.

The game is made by former crytek developers who worked on the cryengine so for the game they are using lumberyard fork which currently doesn't support linux exports. The developer is working on porting the renderer and libraries to get steam deck certified and native linux version with it after the game's release.

Relevant discussion on:
* shatterline discord with the devs https://discord.com/channels/750704373113552976/1022511212661067857 
* with frogs https://discord.com/channels/853130811581530142/855613414584221706/1024017540789305414

Proposed long term solution for issues alike is to fix vkd3d-shader used in the d3dcompiler in wine to compile HLSL shaders without the need to use 3rd party d3dcompiler_47 and mentioned RIO API.

---

If you like the game then consider buying the 5 USD/EUR trinket in the game to support the development and work on linux port and proton compatibility

**Comment by DomiStyle** [comment]
> If you like the game then consider buying the 5 USD/EUR trinket in the game to support the development and work on linux port and proton compatibility

I would but no in-game purchases because the Steam overlay only appears on the EAC launcher window and is not present ingame. Same bug as in Elden Ring.

(Game is indeed working fine though after switching the prefix to win7)

**Comment by Kreyren** [comment]
reportedly the solution is to use `gamescope` or worst case scenario use GeForceNOW to buy things in the game

If anyone has any relevant info then let me know

**Comment by gravit22** [comment]
Hello. The game works fine, but I can't join any online games. Somebody knows how to fix that issue? 
Here is what I get when I try to join:
![image](https://user-images.githubusercontent.com/33228017/192777763-69126d49-1f6f-415e-b5a3-3af3d153dfe1.png)
Session-09-25

**Comment by italoghost** [comment]
@gravit22 make sure to set wine to Windows 7.

**Comment by gravit22** [comment]
> @gravit22 make sure to set wine to Windows 7.

@italoghost yeah, that worked. Thank you.

**Comment by xpander69** [comment]
The Devs have updated the game. It now works out of box with no protontricks required.
The only issue remaining is the ingame purchases not working due to Steam Overlay not working in game.

**Comment by spaghetticodez** [comment]
Steam overlay is the only issue left at the time of writing this, any normal workaround?

Thank you all who made this possible - i get better results in pop / fedora than in w10 :)

EDIT: Gamescope crashes with Nvidia so there is no workaround for me atm, if anyone got more tips, let me know.

**Comment by Kreyren** [comment]
The devs updated the game to give us more usable error messages to reduce some guess work if you get `kicked by anti-cheat` then try to verify game files in steam as that seems to be a common issue after an update.

https://help.steampowered.com/en/faqs/view/0C48-FCBD-DA71-93EB

**Comment by FirephoenixX02** [comment]
Game works fine also with mutliplayer but it utilizes my GPU in multyplayer matches sometimes not enough. GPU usage goes from 90-100% (stable fps) to 10-20% (Unplayable FPS) and then immediately back up to 90-100%. Tried with Proton Stable, Proton Experimental, Proton-GE7-18, Proton-GE7-43. Same problem everywhere.

**Comment by Kreyren** [comment]
@FirephoenixX02 I noticed that too, from looking at it seems to be caused by game compiling shaders so the issue should fix itself the more you play?

**Comment by chtaube** [comment]
On my Arch Linux system, starting the game gives the Shatterline splash screen, and after a while a warning message pops up:

![Shatterline-Screenshot_2023-03-05_15-40-38](https://user-images.githubusercontent.com/929763/222967894-192b86b4-2bc4-448b-a3df-a7556dac664e.png)

Pressing OK closes the window, and while I see the game process is still running, the game never actually starts. Video card is a Radeon RX 6700 XT.

**Steam System Information**: https://gist.github.com/chtaube/db16ccae1068e42d869e5949c930c862

Proton LOG files:
- with Proton experimental: https://gist.github.com/chtaube/9eceee0b4f441c57e8e95243babf3319
- with GE-Proton 7.49: https://gist.github.com/chtaube/416db6cccf11cc38793abd0a787ed4b3

**Comment by DarkestByte** [comment]
@chtaube the same problem seems to exist with some windows users aswell, so it's likely not related to proton. However I did a forum post where they fixed it by setting compatibility to Windows 8 (https://forums.newworld.com/t/unsupported-video-card-detected/261298/3 New World also uses the Lumberyard engine so results should be the same)

Try using Protontricks to set the windows version to 8 or higher and see if that works.

**Comment by kisak-valve** [comment]
Hello @chtaube, `info:    Device name:     : Null hardware (RADV NAVI10)` in your Proton log might be related to the all zero vendor / device id in the error message.

Ignoring that, https://gist.github.com/chtaube/db16ccae1068e42d869e5949c930c862#file-gistfile1-txt-L299 / https://gist.github.com/chtaube/db16ccae1068e42d869e5949c930c862#file-gistfile1-txt-L748-L754 and `info:    Driver: 2.0.249` (https://gist.github.com/chtaube/9eceee0b4f441c57e8e95243babf3319#file-steam-2087030-experimental-log-L1010) tells us that the game is using 32 bit AMDVLK. Can you check how the game behaves with your system configured to use 32 bit mesa/RADV?

**Comment by chtaube** [comment]
@kisak-valve Ok, I found the culprit: _amdvlk_ was not installed (anymore?) on my system. Installed the package and restarted Steam and now Shatterline is running. Did a quick check and multiplayer seems to be working, too.

Thanks for pointing me into the right direction.

**Comment by kisak-valve** [comment]
[REGRESSION] - Shatterline Doesn't Work In Proton 8

Issue transferred from https://github.com/ValveSoftware/Proton/issues/6716.
@ZaxLofful posted on 2023-04-25T05:05:33:

# Compatibility Report
- Name of the game with compatibility issues: Shatterline
- Steam AppID of the game: 2087030

## System Information
- GPU: NVIDIA Corporation NVIDIA GeForce RTX 3080/PCIe/SSE2
- Driver/LLVM version: 4.6.0 NVIDIA 515.105.01
- Kernel version: 5.19.0-40-generic
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/ZaxLofful/12c0287b8b69618707851128a99ea444
- Proton version: 8.0-1

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.



[steam-2087030.tar.gz](https://github.com/ValveSoftware/Proton/files/11318696/steam-2087030.tar.gz)

## Symptoms 

Only splash screen launches and then it closes without fully launching.

## Reproduction

Use Proton 8.0-1, switching to 7.0-6 makes it work again.

**Comment by alasky17** [comment]
@ZaxLofful This regression is fixed with experimental-8.0-20230505 :)

**Comment by zaxlofful** [comment]
@alasky17 : Is there a link to the PR that fixed it and what the problem was?

**Comment by alasky17** [comment]
@ZaxLofful Commit that fixed the problem was https://github.com/ValveSoftware/wine/commit/7035bf12c55228dd57a3787132f3c34654128871

**Comment by jorgicio** [comment]
Since some recent updates, the game got broken and the Shatterline.exe binary couldn't found by the anticheat. Finally there's a workaround: install it in the same partition as the Steam files (and the EAC Runtime). Got it working with Proton Experimental (bleeding edge).

**Comment by kivimango** [comment]
I'm experiencing issues when i switched to Proton Experimental: i cannot play the Expedition game mode: the loading screen shows twice, then i transferred back to the main menu.The other game mdoes are working as expected.
With Proton 7.0.2 everything is fine.

**Comment by kisak-valve** [comment]
Hello @kivimango, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

A Proton dev might also be interested in how the game behaves with Proton 8.0.

**Comment by spaghetticodez** [comment]
Not sure if others have similar issue, but my game always crashes 2nd loading of map - pve/pvp. Run a game, works without issue, end the run and re-queue and always crashes while loading the next map.

Same with GE, 8.0.4 and experimental (bleeding-edge).

[steam-2087030.log](https://github.com/ValveSoftware/Proton/files/13744417/steam-2087030.log)

[steam-2087030_exp.log](https://github.com/ValveSoftware/Proton/files/13744628/steam-2087030_exp.log)

**Comment by sa-mendez** [comment]
> Not sure if others have similar issue, but my game always crashes 2nd loading of map - pve/pvp. Run a game, works without issue, end the run and re-queue and always crashes while loading the next map.
> 
> Same with GE, 8.0.4 and experimental (bleeding-edge).
> 
> [steam-2087030.log](https://github.com/ValveSoftware/Proton/files/13744417/steam-2087030.log)
> 
> [steam-2087030_exp.log](https://github.com/ValveSoftware/Proton/files/13744628/steam-2087030_exp.log)

Yes, this is happening to me as well.  It has been happening for several months now

**Comment by Abduiwahb** [comment]
The new season update broke the game for Proton. It just leads to an infinite loading. 
I sent an email to Shatterline support and they're working on it
![image](https://github.com/ValveSoftware/Proton/assets/132616976/114f8ea7-4df0-4d1a-b10a-9033901fc9f5)
[steam-2087030_hotfix.log](https://github.com/ValveSoftware/Proton/files/14212546/steam-2087030_hotfix.log)
[Game.log](https://github.com/ValveSoftware/Proton/files/14212548/Game.log)

[steam-2087030_exp.log](https://github.com/ValveSoftware/Proton/files/14212545/steam-2087030_exp.log)

**Comment by AJuujarvi** [comment]
@spaghetticodez This should now be fixed in the latest Proton Experimental release.

**Comment by lordcirth** [comment]
Trying to join a match results in "kicked by anticheat".

**Comment by Kreyren** [comment]
> Trying to join a match results in "kicked by anticheat". -- @lordcirth (https://github.com/ValveSoftware/Proton/issues/6165#issuecomment-2185315719)

Try to verify the game's integrity in steam and if it doesn't work then review https://github.com/ValveSoftware/Proton/issues/6165#issuecomment-1258731897 and give us compatibility report with the log from EOS, it should look like this when there is an inconsistency with the game directory:

> <2022-09-26T22:09:54:789+02> [Warning] (Game) - [EOSCore_Windows::EOSSDKLoggingCallback] LogEOSAntiCheat: [AntiCheatClient] [PollStatusInternal] Client Violation with Type: 5 Message: Unknown file version (Bin/CrashRpt1403.dll), len: 44

**Comment by alasky17** [comment]
@lordcirth Joining matches is now working for me.  Could you please retest and see if you still get a message about anti-cheat?

**Comment by lordcirth** [comment]
> @lordcirth Joining matches is now working for me. Could you please retest and see if you still get a message about anti-cheat?

Sorry, didn't see this until now (way too many notifications) and the game is dead now. Thanks.

**Comment by Kreyren** [comment]
>[...] and the game is dead now. Thanks. -- @lordcirth (https://github.com/ValveSoftware/Proton/issues/6165#issuecomment-2467191642)

It's not dead, they are moving from Early Access/Beta Phase to Full Release as they finally decided on which publisher and monetization model to use.

![image](https://github.com/user-attachments/assets/e101fd76-cc28-4b5a-9769-55e88871e677)

The game won't let you into an online match unless it's able to verify the signatures of **ALL** game files which is likely your case (thus the suggestion in https://github.com/ValveSoftware/Proton/issues/6165#issuecomment-2186014531) as either you or your system must have changed something for the anti-cheat to fail it's checks -> Verify the game integrity and it should work again.

Note that on Linux under proton the game has much stricter anti-cheat than on Windows as the Linux's design enables for a deeper integration with the OS and the kernel. On Windows it doesn't seem to care as much about file integrity and if you prefer to play via Epic Games Store then there is an option of https://heroicgameslauncher.com which is an open-source launcher made by the community which should handle these things for you as well as Steam.

Relevant to the issue: To show that the game is not dead with ongoing development to support Linux.

**Comment by AustinPowers1935** [comment]
@Kreyren It is dead on Linux. They moved to egs and does not support anti-cheat on Linux anymore. There is no single clue they are gonna support Linux.

**Comment by Eli2H** [comment]
Yeah I've tried to get it to work, it runs very well with heroic launcher (much better than on Windows actually), even epic online services work, but joining an online match gets me kicked by anticheat in seconds...
If anyone has a working setup, please let me know.

**Comment by Kreyren** [comment]
I talked with the developer and they told me the following:

![image](https://github.com/user-attachments/assets/13e2c013-4e88-4769-934c-db7a777b6724)

The Linux Anti-Cheat Module is active and was never disabled. The EAC Driver initialization returns fail when running under Heroic, but is not responding on the client side for server EAC request which leads to the kick for them to conclude that this is not an error on their side as they used the same code before and on Steam deck. It seems that using EGS as user base platform blocks it from using EAC.

CC @kisak-valve Could you look into this?

**Comment by AustinPowers1935** [comment]
@Kreyren Honestly I do not think Valve would be interested in spending time on this issue. The game is not on Steam anymore. Proton is a fork of wine designed to work specifically with the Steam. Therefore this issue is out of scope for Proton and Valve devs. Why would they care whether EGS exclusive is working on Linux or not?

**Comment by Kreyren** [comment]
From what i understood for Valve it's important that games and everything works in general on Linux due to the existential threat of Xbox/Microsoft Store for this to be in the scope while the devs still have steam developer account and i assume will most likely release the game on steam later assuming the trend with other EGS exclusives e.g. Fall Guys due to the publisher's contract to finance the game's development and marketing.

Plus afaik from the developer's side this is pre-defined way in lumberyard to enable anti-cheat so addressing this will likely be beneficial for other lumberyard games and their linux support.
